### PR TITLE
New article: Santa's Helpers Embrace the Shell

### PR DESCRIPTION
**Pull request by s1037989**
Let me know what I'm doing wrong with my first Advent PR!

### CODE REVIEW

**Comment by oalders** [comment]
@thibaultduponchelle maybe don't merge this after your review as I'm realizing it would generally be good to have 2 reviewers per article.

**Comment by thibaultduponchelle** [comment]
@oalders Ok let’s use this policy 👍🏼

**Comment by thibaultduponchelle** [comment]
I'm going to check code snippets execution and read a second time the article but we really need a native English speaker to review it.

Any volunteer?

**Comment by openstrike** [comment]
I (native English speaker) can review it for the prose if you want?

**Comment by thibaultduponchelle** [comment]
> I (native English speaker) can review it for the prose if you want?

Oh yes please do 🙏🏼

File: 2022/incoming/santas-helper-embrace-the-shell.pod [hunk 1]
```diff
@@ -0,0 +1,141 @@
+Author: Stefan Adams <stefan@adams.fm>
+Title: Santa's Helper Embrace the Shell
+Topic: Command Line
+
+=encoding utf8
+
+=head1 Santa's Helpers Embrace the Shell
+
+Santa's workshop is very busy and everyone on staff is expected to contribute beyond just their traditional roles. Even Santa's reindeer take on additional tasking throughout the year.
```
Comment (openstrike) [inline 2022/incoming/santas-helper-embrace-the-shell.pod:9]: **tasks** would be better than tasking here.

File: 2022/incoming/santas-helper-embrace-the-shell.pod [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+The elves confirmed that Tom, who is naughty, did not get added to Santa's route, but they also noticed that Dick, who is nice, is missing from the list! The reindeers with their poor JSON parser almost caused Dick to get missed! D:
+
+The elves knew this was not a sustainable approach. They decided that they must adopt better tooling to avoid any tragedies with incorrectly skipping a delivery for a child on Christmas day.
+
+=head2 The elves find C<jp>
+
+The elves discovered a new command line tool for parsing these JSON files and it allows the use of pointers to drill down to the specific data of interest.  This new tool is called C<jp> -- for JSON Pointer.
+
+The key features for C<jp> -- made possible by L<Mojo::JSON|https://docs.mojolicious.org/Mojo/JSON> from the wonderful L<Mojolicious|https://mojolicious.org> framework -- are that the elves can use a simple JSON pointer to get at any specific data in their JSON data structure, but then also use Perl data structures to apply logic for ensuring that they extract exactly the right data and not accidentally route Santa to someone on his naughty list.
```
Comment (openstrike) [inline 2022/incoming/santas-helper-embrace-the-shell.pod:34]: "to get at any specific data in their JSON" - I guess this means any specific data set in which case perhaps just explicitly say "data set" here.

File: 2022/incoming/santas-helper-embrace-the-shell.pod [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+With a little Perl, C<jp> transforms into a powerful command line tool that Santa and the elves can count on:
+
+    $ ./jp -p -r '/children/\d+' -E '$_->grep(sub{$_->jp("/list") eq "nice"})' /name /position/0 /location/0 /position/1 /location/1 < advent.json 
+    /children/1     Dick    40.7565323      -73.9904037
+    /children/2     Harry   40.7411595      -74.0097167
+
+=head2 Conclusion
+
+C<jp> is a very new tool and the elves are the first in history to discover its existence. They understand there may be bugs or additional functionality that it lacks, and they are excited to make contributions to improve C<jp> and share those improvements with all the children and everyone else in the world. They have confidence in the tool and their ability to make contributions because it is pretty well documented and comes with a very easy to use test suite that already tests most of C<jp>'s functionality.
```
Comment (openstrike) [inline 2022/incoming/santas-helper-embrace-the-shell.pod:53]: "They understand that there may be" is missing "that"

**Review by openstrike — COMMENTED** [review]
This article reads really well. I have made a few very minor recommendations but would be happy enough to merge as-is if others want to proceed.

**Comment by thibaultduponchelle** [comment]
I'm good with this article. Tested the code snippets 👍🏼 

I have 2 remarks not related to the content of the article but `jp` tool:
1. I think it should be uploaded to CPAN
2. You could add a `cpanfile` (e.g. for installing `Data::Walk::More`)

I going to merge it now.

Thank you a lot @s1037989 let's time to start writing another article? 🤣

**Review by thibaultduponchelle — APPROVED** [review]

**Comment by thibaultduponchelle** [comment]
@openstrike Can you please "approve" this PR so I can merge it (mandatory 2 approvals)

**Review by openstrike — APPROVED** [review]
Approved.

**Pull request merged**